### ISSUE TITLE
FACT meeting workshop

### ISSUE THREAD

**Issue by naomitress** [ocean-tracking-network/otn-workshop-base#85]
todos:
- [x] fork otw repo
- [x] include reporting workshop modules
      - https://ocean-tracking-network.github.io/otn-workshop-base/04-r-telemetry-report-import/index.html
      - https://ocean-tracking-network.github.io/otn-workshop-base/05-r-telemetry-report-array/index.html
      - https://ocean-tracking-network.github.io/otn-workshop-base/05-r-telemetry-report-array/index.html
- [x] cut out sections other than FACT network formats

**Comment by naomitress** [comment]
this needs to be done in December, 2023

**Comment by naomitress** [comment]
@CaitlinBate hopefully this will be staged for you for January work

**Comment by CaitlinBate** [comment]
@jackVanish please keep the intro to R stufd (FACT only) but label them 'background'

**Comment by jackVanish** [comment]
Workshop here: https://ocean-tracking-network.github.io/2024-fact-meeting-workshop/
Repo here: https://github.com/ocean-tracking-network/2024-fact-meeting-workshop

Here's the post-fork checklist, I've done a few things already but we can all chip in

**gh-pages** --> create a new branch to make these changes. use `git checkout -b ＜new-branch＞`

- [ ] - edit the code within each episode to remove the lines that apply to different Nodes
- [ ] - edit episode 00 code to remove any packages we will not be discussing
- [ ] - remove entire lessons we're not teaching and re-number the episodes accordingly
- [ ] - edit index.md to include an accurate summary of the workshop
- [ ] - edit config.yml to have accurate title
- [ ] - edit README.md
- [ ] - remove excess files from Resources folder
- [ ] - edit setup.md, including link to git repo (two places to change this) and remove extra packages we dont need
- [ ] - edit Resources/install_instructions.docx, including link to git repo (two places to change this) and remove extra packages we dont need


**master** --> create a new branch to make these changes. use `git checkout -b ＜new-branch＞`
- [ ] - remove excess Data folders
- [ ] - remove excess Code folders
- [ ] - edit README.md
- [ ] - edit setup/setup.R, including link to git repo (two places to change this) and remove extra packages we dont need
- [ ] - edit setup/install_instructions.docx, including link to git repo (two places to change this) and remove extra packages we dont need

**Comment by jdpye** [comment]
OK, given the layout of what Joy asked for and what we can deliver, we can add to what we already have in that fork a 'Discussion of Report Metrics / Brainstorming sort of section, where we let the users lead us to some of the things they want to build out about their data.

And then Joy originally wanted to cap it off with a GitHub piece, but that can be lecture-style and short if we prefer. Just how to be a good GitHub citizen. I have an old talk I gave on this subject in the last few slides of this talk from ICFT 2019: https://docs.google.com/presentation/d/1p7tqHmxyfmznALj4y0OHAQwdoKUPHg5F/edit?usp=sharing&ouid=108426455008350886281&rtpof=true&sd=true 

but basically something short about why to get started in Git, and how to be a good Gitizen seems reasonable, if Caitlin or Shannon are ok leading a discussion like that? Maybe use Robert and miscYaps as an example?

**Comment by slandovs** [comment]
are workshop attendees supposed to just use the setup instructions in the R script, or refer to the install_instructions.docx? I didn't review this file and am wondering if I should?

**Comment by slandovs** [comment]
reviewing “BASE_1_Intro_to_R_for_Telemetry_Reporting.R”:

- lines 2/3 need to be updated for the FACT workshop
- line 116 needs to be changed to "tqcs_matched_detections_2010.csv" and then the name it was assigned to changed from here down
- line 157 catalognumber needs to be updated to one from the correct file, I used "TQCS-1049258-2008-02-14"
- line 175 needs to be changed to "tqcs_matched_detections_2011.csv" and then the name it was assigned to changed from here down
- line 237 catalognumber needs to be updated to one from the correct file, I used "TQCS-1049258-2008-02-14" again
- line 249 needs to be changed to "teq_qualified_detections_2010_ish.csv" and then the name it was assigned to changed from here down
- line 250 needs to be changed to "teq_qualified_detections_2011_ish.csv" and then the name it was assigned to changed from here down
- line 259 needs to be changed to file "TEQ_Deployments_201001_201201.xlsx" and then remove the sheet = and skip = because they aren't needed for this file
- line 264 need to be changed to file "TQCS_metadata_tagging.xls" and then remove the sheet = and skip = because they aren't needed for this file
- lines 267-269 should be updated to include details on the sheet = and skip =  options, in case people do need them with their own files later
- I got stuck on line 280 because I don't think we have an equivalent file in the data folder? @jackVanish

**Comment by jackVanish** [comment]
Shannon, I'm kicking myself because I made a mistake here. I thought the BASE lesson contained the node-agnostic intro-to-R lessons and the day1_FACT code was the FACT-specific code for later lessons. I checked now and I realized I misremembered. The entire day 1 lesson is in day1_FACT, which should contain all of what you need to run. BASE should not be in the repo, and I'll delete in in my next commit. 

Caitlin or Naomi can scare me straight but I think day1_FACT_node is all you need to test, and that should resolve the filename issues you've been having. Sorry for wasting your time on that one. 

Regarding install_instructions.docx, that has a few extra installation tips for things like RStudio itself or anything else that isn't an R-package. For now, do not worry about it, I will update it once I know all of the above works.

**Comment by slandovs** [comment]
reviewing day1_FACT_Node.R:

- lines 253-260 gives error "Error: Stamen map tiles are now hosted by Stadia Maps, use `get_stadiamap()`". I changed to get_stadiamap, which also required maptype to be changed to ''stamen_terrain_background'' but then got an error stating an API key is needed "Error in `get_stamen_url()`:
! Stadia Maps requires an API key; see `ggmap::register_stadiamaps()`"

**Comment by jackVanish** [comment]
Awful. Sick to death of Stadiamaps. Anyway. I've put a registry key in the setup and changed the code to refer to get_stadiamaps rather than get_stamenmaps. I've also opened a ticket in this repo for us to figure out a longer-term solution, whether that's teaching leaflet or something else other than ggmap, or getting an OTN-proper API key. Anyway, pull the latest version of master (`git pull master`) and run the new lines in setup.R and you should be good to keep rolling.

**Comment by slandovs** [comment]
- line 420: "stamen_terrain-background" needs to be changed to "stamen_terrain_background" (hyphen should be an underscore)
- everything ran without rgeos, so that can be removed from the setup! 
- I still have to read through the website instructions and see if they would be clear enough for someone to use without the instructor!

**Comment by jackVanish** [comment]
OK, good to know. That's enough to feel good about making those changes across the setup files. I'll get on that straightaway.

**Comment by jackVanish** [comment]
Changes have been made to the master branch. 

Shannon's testing strongly suggests that attendees will need neither GDAL nor rgeos so I've removed those from the install instructions in the master branch. I'll do the same in GH-pages as well.

**Comment by slandovs** [comment]
reviewed the web instructions and have a few minor notes. Overall, I think its very clearly written if anyone were to go through it without the instructor! 

- Background - Intro to R: in the code block under "Setting up R" there is a comment specifically to instructors - do we want this here?
- Background - Starting with Data Frames: is separated by node, should FACT be the only one included on this page? 
- Background - Intro to Plotting: same as above, separated by node but do we only want FACT?
- Background - Intro to Plotting: under "Data manipulation" heading, in the second code block, it says "#all dets in/after October of 2016", this should be 2010 instead
- Background - Intro to Plotting: under "Dealing with datetimes" heading, in the last paragraph there is the typo "contianing"
- Telemetry Reports - Imports: there is the typo "regitered" and then in the last sentence "comfirm"
- Telemetry Reports for Array Operators: the first code block needs to be updated to include the change to stadiamaps instead of stamenmap
- Telemetry Reports for Tag Owners: the first code block needs to be updated to include the change to stadiamaps instead of stamenmap
- Telemetry Reports for Tag Owners: under "Other example plots" header, in the code block there is a question mark at the end of one of the comments - should it be there? The comment is #to size the dots by number of detections you could do something like: size = (log(length(animal)id))?

**Comment by jackVanish** [comment]
OK, here's where we're at: across both major branches, I've harmonized the setup and install instructions. Shannon's tested the package set that I included, so we should be good to go on that front- no messy GDAL or rgeos installs, should be smooth sailing on that front (although more testing is always welcome). There are a few unticked boxes, I'll explain below:

- I removed actel, remora, and pathroutr from episode 0, and after checking, I'll go back and remove glatos as well, since it doesn't seem to be part of these lessons (feel free to scare me straight on that one). I've left the box unchecked just so the folk giving the lessons can take a peek and make sure it's all copacetic before they sign off on it. So episode 00 just needs some review from the lesson-givers.
- index.md in GH-pages I've left for whoever's giving the lesson to edit with an appropriate description. They'll know better than I, and so it makes more sense to let them put a paragraph or two in there. 
- I took a bunch of things out of the Resources folder that I knew were unnecessary (to do with YAPS and GAMS and suchlike) but I figured I wouldn't tick this one off until the lesson-givers had taken a look and gotten rid of anything they didn't need. 

So, to summarize, episode 00, index.md, and the Resources folder in gh-pages need a little review from whoever's giving the lesson to make sure they're locked in. I also have to go and delete the node-specific lessons other than FACT from the early 'background' lessons, since I forgot to do that earlier (no biggie). Beyond those, this should be wrapped up and ready to shoot, although I would, as always, encourage more testing just to make sure I haven't missed anything.

**Comment by jackVanish** [comment]
I've now taken ACT, GLATOS, and MIGRAMAR from episodes 2 and 3.

**Comment by CaitlinBate** [comment]
> Changes have been made to the master branch.
> 
> Shannon's testing strongly suggests that attendees will need neither GDAL nor rgeos so I've removed those from the install instructions in the master branch. I'll do the same in GH-pages as well.

This ticket is in otn-workshop-base --- were all changes (typos) fixed in both the workshop itself and otn-workshop-base? the location of this ticket vz the work happening is confusing to me

**Comment by CaitlinBate** [comment]
lessons on the website still refer to Stamen maps - the code on the webpage needs to be updated with all the fixes that are in the .R file, as identified above.

**Comment by slandovs** [comment]
@CaitlinBate I believe that is what I just did? I did fixes on the .md files for the stamen map stuff!

**Comment by CaitlinBate** [comment]
feeling good about closing this one, and opening new tickets in here if we need to https://github.com/ocean-tracking-network/2024-fact-meeting-workshop/issues
